### PR TITLE
feat(ui5-library-inquirer): Support caller prompt module integration

### PR DESCRIPTION
**Pull request by IainSAP**
https://github.com/SAP/open-ux-tools/issues/1444

- Extends `prompt()` signature to allow caller to pass prompting module reference (`InquirerAdaptor`) thereby allowing the calling module to integrate the prompts more easily. Especially useful for calling Yeoman generators when running in a UI environment rather than CLI.

- Adds tests to cover

### CODE REVIEW

**Review by devinea — APPROVED** [review]
Changeset ✅ 
Code changes and tests look good.

File: packages/ui5-library-inquirer/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     },
     "devDependencies": {
         "@types/inquirer-autocomplete-prompt": "2.0.1",
-        "@types/inquirer": "8.2.7"
+        "@types/inquirer": "8.2.6"
```
Comment (tobiasqueck) [inline packages/ui5-library-inquirer/package.json:43]: I am curious, why did you switch to an older version of the types?

Comment (IainSAP) [reply packages/ui5-library-inquirer/package.json:43]: It was misaligned wit the version in use : https://github.com/SAP/open-ux-tools/pull/1456/files/0ed37512b559b6d6f2af4216749e62c5a2f2926c#diff-4b021372a31ed3cd94bdc75d23e3bf011cba40e39479b52518989610f16eb8f2R38

Comment (IainSAP) [reply packages/ui5-library-inquirer/package.json:43]: It was misaligned with the inquirer version being used.

**Review by tobiasqueck — APPROVED** [review]
* changeset ok
* code straight forward and adheres to coding guidelines
* good test coverage
* very helpful enhancement of the README (thanks!)
* did NOT test it locally

**Review by devinea — APPROVED** [review]
re-approving.

**Pull request merged**